test/unit/test_env_variable.py

```diff
@@ -0,0 +1,215 @@
+import pytest
+import os.path
+import tempfile
+from pathlib import Path
+from unittest import mock
+
+import luigi
+
+from exasol_integration_test_docker_environment.lib.base.luigi_log_config import LOG_ENV_VARIABLE_NAME, get_log_path
+from exasol_integration_test_docker_environment.lib.api.common import generate_root_task, run_task
+from exasol_integration_test_docker_environment.lib.base.dependency_logger_base_task import DependencyLoggerBaseTask
+from exasol_integration_test_docker_environment.lib.config.build_config import build_config
+
+
+@pytest.fixture
+def set_tempdir(tmp_path):
+    path_old = os.getcwd()
+    os.chdir(tmp_path)
+    yield tmp_path
+    os.chdir(path_old)
+
+
+@pytest.fixture()
+def mock_settings_env_vars():
+    with mock.patch.dict(os.environ, {}):
+        yield
+
+
+def default_log_path(job_id):
+    return Path(build_config().output_directory) / "jobs" / job_id / "logs" / "main.log"
+
+
+class TestTask(DependencyLoggerBaseTask):
+    x = luigi.Parameter()
+
+    def run_task(self):
+        self.logger.info(f"Logging: {self.x}")
+        self.return_object({"job_id": self.job_id, "parameter": self.x})
+
+
+def run_n_simple_tasks(task_number):
+    NUMBER_TASK = task_number
+    task_id_generator = (x for x in range(NUMBER_TASK))
+    tasks = []
+
+    def create_task():
+        return generate_root_task(task_class=TestTask, x=f"{next(task_id_generator)}")
+
+    for j in range(NUMBER_TASK):
+        output = run_task(create_task, workers=5, task_dependencies_dot_file=None, use_job_specific_log_file=True)
+        jobid = output["job_id"]
+        log_path = get_log_path(jobid)
+        tasks.append({"jobid": jobid, "log_path": log_path, "in_parameter": output["parameter"]})
+
+    return tasks
+
+
+def use_specific_log_file(task_creator, temp_dir, test_name):
+    log_path = Path(temp_dir) / (test_name + ".log")
+    os.environ[LOG_ENV_VARIABLE_NAME] = str(log_path)
+    run_task(task_creator, workers=5, task_dependencies_dot_file=None, use_job_specific_log_file=True)
+    return log_path
+
+
+def test_var_not_set(set_tempdir):
+    """
+    Integration test which verifies that not setting the env var LOG_ENV_VARIABLE_NAME works and uses the
+    default log path
+    """
+    tasks = run_n_simple_tasks(1)
+
+    log_path = tasks[0]["log_path"]
+    assert log_path == default_log_path(tasks[0]["jobid"])
+    assert log_path.exists()
+    assert os.path.isfile(log_path)
```

Comment:
```suggestion
    assert log_path.is_file()
```

Reply:
use os package, only if there is no option in pathlib for the function

Reply:
I would already, suggest to write a matcher object for the logpath, which runs the conditions of the asserts in `__eq__` method, this also includes the content asserts. This way you have only one assert per test and not so much code duplication through the tests. 
Here some examples:

- https://github.com/exasol/script-languages-container-ci/blob/2d6a535c44ff16471ac8a44e7bf0026f62823f34/test/matchers.py#L5
- https://github.com/exasol/transformers-extension/blob/0b8dc06c4fc39f902eaf83c8e021f4dd1c59e58a/tests/unit_tests/udfs/output_matcher.py#L18